### PR TITLE
Added genesis checker

### PR DESCRIPTION
**Pull request by vaibhavchellani**

### CODE REVIEW

**Comment by vaibhavchellani** [comment]
WIP: Testing underway.

File: cmd/heimdalld/main.go [hunk 1]
```diff
@@ -223,6 +226,82 @@ func InitCmd(ctx *server.Context, cdc *codec.Codec) *cobra.Command {
...
+			}
+
+			// validate validators
+			validators := genesisState.StakingData.Validators
+			for _, v := range validators {
+				val, err := contractCaller.GetValidatorInfo(v.ID)
+				if err != nil {
+					return err
+				}
+				if val.VotingPower != v.VotingPower {
```
Comment (jdkanani) [inline cmd/heimdalld/main.go]: Please use the`GetPowerFromAmount()` method to get the amount before comparing to genesis state validator power. 

Check method: https://github.com/maticnetwork/heimdall/blob/38db2c8598ffce6594eb650565d089ff82703691/helper/util.go#L640-L647

Comment (jdkanani) [reply cmd/heimdalld/main.go]: I think you don't it since ValidatorInfo is doing before sending it from the contract.

File: staking/handler.go [hunk 2]
```diff
@@ -91,7 +91,7 @@ func HandleMsgValidatorJoin(ctx sdk.Context, msg MsgValidatorJoin, k Keeper, con
 		ID:          validator.ID,
 		StartEpoch:  validator.StartEpoch,
 		EndEpoch:    validator.EndEpoch,
-		VotingPower: validator.VotingPower,
+		VotingPower: types.GetValidatorPower(validator.VotingPower),
```
Comment (jdkanani) [inline staking/handler.go]: No need to do that since validator already has voting power.

Comment (vaibhavchellani) [reply staking/handler.go]: Yes, I found 2 functions, was planning on removing one.

**Comment by jdkanani** [comment]
Tested and fixed

**Pull request merged**